ledger/src/advance.rs

```diff
@@ -147,28 +147,28 @@ where
         }
 
         // Split off a chunk of the candidate solutions.
-        let candidates_chunk = if candidate_solutions.len() > chunk_size {
+        let mut candidates_chunk = if candidate_solutions.len() > chunk_size {
             candidate_solutions.split_off(candidate_solutions.len() - chunk_size)
         } else {
             std::mem::take(&mut candidate_solutions)
         };
 
         // Verify the solutions in the chunk.
         let verification_results: Vec<_> = candidates_chunk
-            .into_iter()
+            .iter_mut()
             .rev()
             .map(|solution| {
-                let verified = verification_fn(&solution);
+                let verified = verification_fn(solution);
                 (solution, verified)
             })
             .collect();
```

Comment:
this isn't introduced in this PR, but it doesn't seem that we need this intermediate `collect` into a vector?

Reply:
Fixed with [fa3a03d](https://github.com/AleoNet/snarkVM/pull/2520/commits/fa3a03dd151929a8af4eb879f9fc51499d13e33b). 

However, I believe this entire function can be overhauled/optimized in subsequent PRs. We currently aren't parallelizing verification.